lib/Sema/CSGen.cpp

```diff
@@ -1676,9 +1676,9 @@ namespace {
                                      TVO_CanBindToHole);
       }
       // Diagnose top-level usages of placeholder types.
-      if (isa<PlaceholderTypeRepr>(repr->getWithoutParens())) {
-        CS.getASTContext().Diags.diagnose(repr->getLoc(),
-                                          diag::placeholder_type_not_allowed);
+      if (auto *ty = dyn_cast<PlaceholderTypeRepr>(repr->getWithoutParens())) {
+        auto *loc = CS.getConstraintLocator(locator, {LocatorPathElt::PlaceholderType(ty)});
+        CS.recordFix(IgnoreInvalidPlaceholder::create(CS, loc));
```

Comment:
Should this be more impactful than default?

Reply:
Hmm, does the impact make a difference for fixes that track something structurally wrong with the AST (in this case the TypeRepr)?

Reply:
Even though all of the solutions are going to have it in this case I think it's worth increasing it to inform future uses maybe just like what happens with other structural fixes...

Reply:
Spoke with Pavel offline, going to keep this as-is for now